core/src/main/scala/kafka/message/MessageSet.scala

```diff
@@ -54,6 +58,22 @@ object MessageSet {
    */
   def entrySize(message: Message): Int = LogOverhead + message.size
 
+  /**
+   * Validate the "magic" values of messages are the same in a compressed message set and return the max timestamp
```

Comment:
We only return the max timestamp of the inner messages if timestampType is createTime.